gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFormat.java

```diff
@@ -95,7 +111,21 @@ static boolean eclipseFormattingEnabled(Project project) {
         return project.hasProperty(ECLIPSE_FORMATTING);
     }
 
+    static boolean palantirJavaFormatterEnabled(Project project) {
+        return project.hasProperty(PJF_PROPERTY);
+    }
+
     static Path eclipseConfigFile(Project project) {
         return project.getRootDir().toPath().resolve(".baseline/spotless/eclipse.xml");
     }
+
+    private static class PalantirJavaFormatterFunc implements FormatterFunc {
+        private final Formatter palantirJavaFormatter =
+                new Formatter(JavaFormatterOptions.builder().style(Style.PALANTIR).build());
```

Comment:
I've made this constructor private, so let's switch to service-loading!

Reply:
yeah, sadly we need to access more internals to do the string -> string conversion, but luckily the other internals are still public :)